src/main/java/duck/IoHandler.java

```diff
@@ -46,45 +45,51 @@ public String typeMessage() {
      *
      * @param taskList The TaskList containing tasks to be displayed.
      */
-    public void display(TaskList taskList) {
+    public String display(TaskList taskList) {
         int count = 0;
         int serial = 1;
-        divider();
+        StringBuilder result = new StringBuilder();
+        if (taskList.size() == 0) {
+            return "There are no tasks in your lists.";
+        }
+        result.append("Here are all the tasks in your list: \n");
```

Comment:
I love how you used StringBuilder in this function, this is definitely something I can learn from you and will try to implement in my code :)